server/models/Classroom.coffee

```diff
@@ -44,13 +45,74 @@ ClassroomSchema.pre('save', (next) ->
     next()
 )
 
-
 ClassroomSchema.methods.isOwner = (userID) ->
   return userID.equals(@get('ownerID'))
 
 ClassroomSchema.methods.isMember = (userID) ->
   return _.any @get('members') or [], (memberID) -> userID.equals(memberID)
 
+ClassroomSchema.methods.generateCoursesData = co.wrap (isAdmin=false) ->
+  # Helper function for generating the latest version of courses
+  query = {}
+  query = {releasePhase: 'released'} unless isAdmin
+  courses = yield Course.find(query)
+  courses = Course.sortCourses courses
+  campaigns = yield Campaign.find({_id: {$in: (course.get('campaignID') for course in courses)}})
+  campaignMap = {}
+  campaignMap[campaign.id] = campaign for campaign in campaigns
+  classLanguage = @get('aceConfig')?.language
+  coursesData = []
+  for course in courses
+    courseData = { _id: course._id, levels: [] }
+    campaign = campaignMap[course.get('campaignID').toString()]
+    levels = _.values(campaign.get('levels'))
+    levels = _.sortBy(levels, 'campaignIndex')
```

Comment:
Uh oh, now we're code reviewing @sderickson code I moved but didn't change :)  Will update.